src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -311,6 +319,75 @@ private void WriteComponentHtml(int componentId, TextWriter output, bool allowBo
         }
     }
 
+    private void AppendWebAssemblyPreloadHeaders()
+    {
+        var assets = _httpContext.GetEndpoint()?.Metadata.GetMetadata<ResourceAssetCollection>();
+        if (assets != null)
+        {
+            var headers = new List<(string? Order, string Value)>();
+            foreach (var asset in assets)
+            {
+                if (asset.Properties == null)
+                {
+                    continue;
+                }
+
+                // Use preloadgroup=webassembly to identify assets that should to be preloaded
+                string? header = null;
+                foreach (var property in asset.Properties)
+                {
+                    if (property.Name.Equals("preloadgroup", StringComparison.OrdinalIgnoreCase) && property.Value.Equals("webassembly", StringComparison.OrdinalIgnoreCase))
```

Comment:
The feature for identifying assets that should be preload seems quite generic. I have introduced here an ability to group such preloads. It would give us possibility in the future to define component or other API that would allow to preload other assets in other scenarios.

Alternatively, I can prefix all the SWA properties with `webassembly-` to make WebAssembly specific.

Reply:
No, I think you did the right thing here.